pkg/apis/v1alpha1/vpc_types.go

```diff
@@ -0,0 +1,96 @@
+/* Copyright � 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+   package v1alpha1
+
+   import (
+	   corev1 "k8s.io/api/core/v1"
+	   metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+   )
+   
+   type VPCStatusCondition string
+   
+   const (
+	   VPCReady    VPCStatusCondition = "Ready"
+	   VPCNotReady VPCStatusCondition = "NotReady"
+   )
+   
+   //+kubebuilder:object:root=true
+   //+kubebuilder:subresource:status
+   //+kubebuilder:storageversion
+   
+   // VPC is the Schema for the VPC API
+   type VPC struct {
+	   metav1.TypeMeta   `json:",inline"`
+	   metav1.ObjectMeta `json:"metadata,omitempty"`
+   
+	   Spec   VPCSpec   `json:"spec,omitempty"`
+	   Status VPCStatus `json:"status,omitempty"`
+   }
+   
+   //+kubebuilder:object:root=true
+   
+   // VPCList contains a list of VPC
+   type VPCList struct {
+	   metav1.TypeMeta `json:",inline"`
+	   metav1.ListMeta `json:"metadata,omitempty"`
+	   Items           []VPC `json:"items"`
+   }
+   
+   // VPCSpec defines VPC configuration
+   type VPCSpec struct {
+	   // DHCP config specified at VPC level is applicable to all the Subnets under the VPC.
+	   // +optional
+	   DHCPConfig DHCPConfig `json:"dhcpConfig"`
+   }
+   
+    type DHCPConfig struct {
+	   // If turned on, the DHCP server will be configured based on IP address type.
+	   // If turned off, then neither DHCP server nor relay shall be configured.
+	   // +optional
+	   // +kubebuilder:default=true
+	   EnableDHCPServer bool `json:"enable_dhcp_server"`
+	   // DNS configuration for VPC.
+	   // +optional
+	   DNSConfig DNSConfig `json:"dnsConfig"`
+   }
+   
+   type DNSConfig struct {
+	   // IP address list of DNS servers.
+	   // +optional
+	   DNSServerIPs []string `json:"dnsServerIPs"`
+   }
+   
+   // VPCStatus defines the observed state of VPC
+   type VPCStatus struct {
+	   Conditions []VPCCondition `json:"conditions"`
+	   // NSX VPC Policy API resource path.
+	   NSXVPCPath string `json:"nsxVPCPath"`
```

Comment:
I discussed this field with Jianjun and Xiaopei, we prefer to change it to `NSXPath`, then all CRDs can use the same field name, which looks more common. What do you think?

Reply:
How about NSXAPIPath or NSXPolicyAPIPath to be clear?

Reply:
I prefer NSXPolicyPath. I see they called NSX Policy Path. @jianjuns